SMLHelper/Assets/HullPlate.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace SMLHelper.V2.Assets
+{
+    using System;
+    using SMLHelper.V2.Crafting;
+    using UnityEngine;
+
+    /// <summary>
+    /// A hull plate item that can be built into the game world.
+    /// </summary>
+    /// <seealso cref="PdaItem" />
+    /// <seealso cref="Spawnable"/>
+    /// <seealso cref="Buildable"/>
+    public abstract class HullPlate : Buildable
+    {
+        /// <summary>
+        /// Gets the <see cref="Texture2D"/> that the hull plate should have
+        /// </summary>
+        /// <returns></returns>
+        public abstract Texture2D GetTexture();
```

Comment:
The base class `Spawnable` from which this class inherits, has a virtual implementation for loading its [icon sprite](https://github.com/SubnauticaModding/SMLHelper/blob/master/SMLHelper/Assets/Spawnable.cs#L131) from a PNG file at run time.

Could this not do the same?  
Could this method have a virtual implementation, also making use of our [ImageUtils](https://github.com/SubnauticaModding/SMLHelper/blob/master/SMLHelper/Utility/ImageUtils.cs), such that it could detect an image file by name?  
If so, allow the texture file to also have a default value that can be overridden.

Reply:
I personally don't like SMLHelper's way of making optional properties required or making required properties optional.

This method will most definitely be used in conjunction with `ImageUtils`, but that doesn't mean that I should turn this into a string property to the path of the file.

Maybe the modder wants to get the texture from an asset bundle, or idk. If they want to load the image from a file, what's stopping them form using `ImageUtils` themselves?

(More on this topic later in the Discord)